cmd/create_test.go

```diff
@@ -90,33 +90,101 @@ func (s *MySuite) TestSetBackendConfig(c *C) {
 		"odor=strong",
 	}
 
-	bp := config.Blueprint{}
-	c.Assert(setBackendConfig(&bp, vars), IsNil)
+	ds := config.DeploymentSettings{}
+	c.Assert(setBackendConfig(&ds, vars), IsNil)
 
-	be := bp.TerraformBackendDefaults
+	be := ds.TerraformBackendDefaults
 	c.Check(be.Type, Equals, "green")
 	c.Check(be.Configuration.Items(), DeepEquals, map[string]cty.Value{
 		"taste": cty.StringVal("sweet"),
 		"odor":  cty.StringVal("strong"),
 	})
 }
 
+func (s *MySuite) TestMergeDeploymentSettings(c *C) {
+	ds := config.DeploymentSettings{
+		Vars: config.NewDict(map[string]cty.Value{
+			"project_id":      cty.StringVal("ds_test_project_id"),
+			"deployment_name": cty.StringVal("ds_deployment_name"),
+		}),
+	}
+
+	bp := config.Blueprint{
+		Vars: config.NewDict(map[string]cty.Value{
+			"project_id":  cty.StringVal("bp_test_project_id"),
+			"example_var": cty.StringVal("bp_example_value"),
+		}),
+	}
+
+	// test priority-based merging of deployment variables
+	mergeDeploymentSettings(&bp, ds)
+	c.Check(bp.Vars.Items(), DeepEquals, map[string]cty.Value{
+		"project_id":      cty.StringVal("ds_test_project_id"),
+		"deployment_name": cty.StringVal("ds_deployment_name"),
+		"example_var":     cty.StringVal("bp_example_value"),
+	})
+
+	// check merging zero-value backends
+	ds = config.DeploymentSettings{
```

Comment:
Do not reuse the namespace `ds` `bp` for subsequent tests. Aka each time you define an object use the `:=` and come up with new naming.

if the same object works for multiple tests then fine to reuse, just not redefine. 

Reason: test is fragil to cases being inserted, harder to trace back to what objects are actually being tested.

Reply:
Please review https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/2224/commits/ee4a2d8ad29a12ff660d7762ea37c0f60d9db13a